ports/microsoft-windows-devices-midi2/portfile.cmake

```
@@ -0,0 +1,120 @@
+set(MINIMUM_WINDOWS_SDK_VERSION "10.0.26100.0")
+
+set(MIDI_SDK_VERSION "${VERSION}")
+set(MIDI_SDK_NUGET_URL "https://github.com/microsoft/MIDI/releases/download/preview-13/Microsoft.Windows.Devices.Midi2.${MIDI_SDK_VERSION}.nupkg")
+set(MIDI_SDK_SHA512 e950cf87ec74df7b8fb8d06c1c09646f5a9f390fa1d19b9906cc79874f52310bd90a80371f9bb089f953794b05d013d602780a5905ba77aa8d8a1a6205d341d8)
+
+message("MIDI2: MIDI SDK Version:   ${MIDI_SDK_VERSION}")
+message("MIDI2: MIDI SDK NuGet URL: ${MIDI_SDK_NUGET_URL}")
+
+# Get the MIDI SDK ---------------------------------------------------------------------------
+# Grab the NuGet package from the official release location
+vcpkg_download_distfile(
+    MIDISDK_ARCHIVE
+    URLS "${MIDI_SDK_NUGET_URL}"
+    FILENAME "Microsoft.Windows.Devices.Midi2.${MIDI_SDK_VERSION}.zip"
+    SHA512 ${MIDI_SDK_SHA512}
+  )
+
+# NuGet files are just zip files, so we extract it here
+vcpkg_extract_source_archive(
+    MIDI_SDK_EXTRACTED_FILES
+    ARCHIVE "${MIDISDK_ARCHIVE}"
+    NO_REMOVE_ONE_LEVEL
+  )
+
+set(MIDI2_WINRT_WINMD_SOURCE "${MIDI_SDK_EXTRACTED_FILES}/ref/native/Microsoft.Windows.Devices.Midi2.winmd")
+
+file(INSTALL 
+    "${MIDI2_WINRT_WINMD_SOURCE}" 
+    DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}"
+  )  
+
+set(MIDI2_WINRT_WINMD "${CURRENT_PACKAGES_DIR}/share/${PORT}/Microsoft.Windows.Devices.Midi2.winmd")
+
+
+set(_MIDI2_HEADERS_ROOT_FOLDER "${CURRENT_PACKAGES_DIR}/include")
+
+# these two bootstrapper files are included in the NuGet package, and are not generated by cppwinrt
+# the sub path they are placed in is consistent with the NuGet package folder structure
+
+set(_MIDI2_HEADERS_INIT_FOLDER "${_MIDI2_HEADERS_ROOT_FOLDER}/winmidi/init/")
+set(_MIDI2_HEADERS_WINRT_FOLDER "${_MIDI2_HEADERS_ROOT_FOLDER}/winrt/")
+
+
+file(INSTALL 
+    FILES
+      "${MIDI_SDK_EXTRACTED_FILES}/build/native/include/winmidi/init/Microsoft.Windows.Devices.Midi2.Initialization.hpp" 
+      "${MIDI_SDK_EXTRACTED_FILES}/build/native/include/winmidi/init/WindowsMidiServicesVersion.h" 
+    DESTINATION "${_MIDI2_HEADERS_INIT_FOLDER}"
+  )
+
+# Find and use the latest SDK, but needs to be a minimum version as defined above.
+vcpkg_get_windows_sdk(WINDOWS_SDK_VERSION)
+
+if (WINDOWS_SDK_VERSION VERSION_GREATER_EQUAL "${MINIMUM_WINDOWS_SDK_VERSION}")
+  message(STATUS "MIDI2: found Windows SDK: ${WINDOWS_SDK_VERSION}")
+else()
+  message(FATAL_ERROR "MIDI2: Need a Windows SDK version that is at least ${MINIMUM_WINDOWS_SDK_VERSION}")
+endif()
+
+configure_file("${CMAKE_CURRENT_LIST_DIR}/microsoft-windows-devices-midi2-config.cmake"
+    "${CURRENT_PACKAGES_DIR}/share/${PORT}/${PORT}-config.cmake"
+    @ONLY
+  )
+
+file(INSTALL 
+    "${CMAKE_CURRENT_LIST_DIR}/usage" 
+    DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}"
+  )
+
+vcpkg_download_distfile(
+    LICENSE_FILE
+    URLS "https://github.com/microsoft/MIDI/raw/30f42326e4ec0072cb6fd846a9b1230149fd4888/LICENSE"
+    FILENAME LICENSE

```

This filename needs to identify the port it's coming from and probably contain the version.

---

Also note that it's installed on line 78. The docs for that say the file ends up here

`${CURRENT_PACKAGES_DIR}/share/${PORT}`

---

The problem is that this is the name taken in the downloads directory. So this ends up taking `C:\Dev\vcpkg\downloads\LICENSE` or similar which is likely to cause conflicts if 2 users do this.

If you think there's no way the version would change it it should still be something like:

```suggestion
    FILENAME microsoft-windows-devices-midi2-LICENSE
```

> Also note that it's installed on line 78. The docs for that say the file ends up here
> 
> `${CURRENT_PACKAGES_DIR}/share/${PORT}`

You can install it as that name because `${PORT}` is in the path, but `${PORT}` isn't in download paths.